### DISCUSSION TITLE
Dev Question: Editor Feature Question - Pane vs Window vs Workspace vs Editor etc.

### DISCUSSION THREAD

**Discussion by solventak** [zed-industries/zed#9763] [Feature Requests]
As I'm working through getting up to speed with the code base I see a lot of references to Workspace, Worktree, Editor, Pane, Pane Group etc.

Is there any place that defines all of these components?  I can't quite grok it yet from the code.

Thanks!

**Comment by SomeoneToIgnore** [comment]
`pub struct/enum` should definitely have a rustdoc on it, I have started to fill the gaps, hopefully that helps: https://github.com/zed-industries/zed/pull/9772

**Reply by solventak** [reply]
This is great.  Thank you so much for the quick reply.  Looking forward to seeing the final result
